app/decorators/physical_server_decorator.rb

```diff
@@ -0,0 +1,14 @@
+class PhysicalServerDecorator < MiqDecorator
+  def img_for_physical_vendor
+    "svg/vendor-#{label_for_vendor.downcase}.svg"
+  end
+
+  def img_for_health_state
```

Comment:
I agree with also having `fonticon` and `fileicon` methods. But the point of moving these other methods closer to the model is that they are more related to PhysicalServer. It's not super clear whose responsibility an image path is, but it's more likely the model's than the QuadiconHelper module. I also agree that having to know the type `@record` isn't great, but most of this code is type-checks on `@record` anyway.